### ISSUE TITLE
Generated config produces incomplete Required string fields

### ISSUE THREAD

**Issue by almereyda** [matrix-org/matrix-appservice-irc#67]
After generation of the appservice configuration YAML, which is to be loaded by the homeserver, the latter complains:

```
matrix_1 | 2016-06-07 13:39:55,498 - synapse.storage.appservice - 238 - ERROR - - Failed to load appservice from '/data/appservice-registration-irc.yaml'
matrix_1 | 2016-06-07 13:39:55,498 - synapse.storage.appservice - 239 - ERROR - - "Required string field: 'id' (/data/appservice-registration-irc.yaml)"
matrix_1 | Traceback (most recent call last):
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/appservice.py", line 215, in load_appservices
matrix_1 |     hostname, yaml.load(f), config_file
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/appservice.py", line 155, in _load_appservice
matrix_1 |     field, config_filename,
matrix_1 | KeyError: "Required string field: 'id' (/data/appservice-registration-irc.yaml)"
matrix_1 | Traceback (most recent call last):
matrix_1 |   File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
matrix_1 |     "__main__", fname, loader, pkg_name)
matrix_1 |   File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
matrix_1 |     exec code in run_globals
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 750, in <module>
matrix_1 |     main()
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 745, in main
matrix_1 |     hs = setup(sys.argv[1:])
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 420, in setup
matrix_1 |     hs.setup()
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/server.py", line 120, in setup
matrix_1 |     self.datastore = DataStore(self.get_db_conn(), self)
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/__init__.py", line 178, in __init__
matrix_1 |     super(DataStore, self).__init__(hs)
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/registration.py", line 29, in __init__
matrix_1 |     super(RegistrationStore, self).__init__(hs)
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/appservice.py", line 39, in __init__
matrix_1 |     hs.config.app_service_config_files
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/appservice.py", line 215, in load_appservices
matrix_1 |     hostname, yaml.load(f), config_file
matrix_1 |   File "/usr/local/lib/python2.7/dist-packages/synapse/storage/appservice.py", line 155, in _load_appservice
matrix_1 |     field, config_filename,
matrix_1 | KeyError: "Required string field: 'id' (/data/appservice-registration-irc.yaml)"
```

and indeed the configured configuration

```
url: 'http://matrix.allmende.io:9999'
as_token: 
hs_token: 
sender_localpart: appservice_irc
namespaces:
  users:
    - exclusive: true
      regex: '@indymedia_.*:.*'
  aliases:
    - exclusive: true
      regex: '#indymedia_.*:.*'
  rooms:
    - exclusive: false
      regex: '#ecobytes:matrix.allmende.io'
```

does not bring [an id](https://github.com/matrix-org/synapse/blob/master/synapse/storage/appservice.py#L150).

What have I done wrongly after filling out the config.yaml respective to my case?

**Comment by almereyda** [comment]
Let's just set a random one https://github.com/matrix-org/synapse/blob/master/synapse/storage/appservice.py#L217-L223

**Comment by Kegsay** [comment]
The requirement for an `id` to be present on Application Services is new. Current `master` does not do this, but `develop` has for some time now: https://github.com/matrix-org/matrix-appservice-irc/blob/develop/lib/main.js#L74

Presently, `develop` is what you should be using and will work for *most* workloads. There's currently some performance issues with it (see #68) which means that for the Freenode x Matrix.org scenario which is a huge firehose of data, things slow down drastically to the point of unusability, which is why `develop` hasn't been merged to `master` yet. When this happens, this issue will go away.

tl;dr - use `develop` for now.

**Comment by almereyda** [comment]
Thanks for pointing this out. I already gained the impression [`develop` is the place to be](https://github.com/vector-im/vector-web/issues/1558#issuecomment-222563455) within the Matrix ecosystem.

**Comment by Kegsay** [comment]
The intention for the bridge is for people to be on `master`, but unfortunately this regression is blocking the release until it is fixed.

**Comment by Kegsay** [comment]
Things got merged; this is now fixed.
